projects/observability/src/shared/components/cartesian/d3/legend/cartesian-legend.ts

```diff
@@ -202,16 +259,57 @@ export class CartesianLegend {
     this.activeSeriesSubject.next();
   }
 
-  private updateActiveSeries(seriesEntry: Series<{}>): void {
-    if (this.isDefault) {
-      this.activeSeries = [];
-      this.activeSeries.push(seriesEntry);
-      this.isDefault = false;
+  private getGroupedSeries(): Series<{}>[][] {
+    if (!this.isGrouped) {
+      return [];
+    }
+
+    const nonHiddenSeries = this.series.filter(series => !series.hide);
+    const groupedSeries: Series<{}>[][] = [];
+    const currentGroup: Series<{}>[] = [];
+
+    nonHiddenSeries.forEach(seriesEntry => {
```

Comment:
Could you please comment or just explain in this PR how exactly this solution is working? A step wise breakup would be great to understand

Reply:
This foreach still doesn't really make sense to me - I would expect us to just group series by `series.groupName ?? series.name`. Once all grouped, render the groups of > 1 with the new style, the groups of 1 as the existing style.

Here, If I had three series and they belong to groups A, B and A respectively, it would break because we're unnecessarily depending on order. Further, having the same wrapper shape for every series allows us to remove all the if/else logic around such as what we have on 281 (I remember a thread started there, but not sure if I missed responding to it - will try to dig that up).